frontend/js/src/listens/ListenCard.tsx

```diff
@@ -118,8 +117,11 @@ export default class ListenCard extends React.Component<
     oldState: ListenCardState
   ) {
     const { listen: oldListen } = oldState;
-    const { customThumbnail } = this.props;
-    const { listen } = this.state;
+    const { listen, customThumbnail } = this.props;
+    if (!isEqual(listen, oldListen)) {
+      this.setState({ listen });
+    }
```

Comment:
Looking at this PR that implements a similar solution, we should probably check that the listen is not falsy just in case:
https://github.com/metabrainz/listenbrainz-server/pull/2639/files#diff-58b7d1f4892674a0d59758a5f774056eb2a32909cfcbf6d7444ee47f17d65e36R123-R125